dataplane/linux/bpf_ep_mgr.go

```diff
@@ -1344,3 +1370,23 @@ func FindJumpMap(ap *tc.AttachPoint) (mapFD bpf.MapFD, err error) {
 	}
 	return 0, errors.New("failed to find TC program")
 }
+
+func (m *bpfEndpointManager) getInterfaceIP(ifaceName string) (net.IP, error) {
+	if ip, ok := m.ifaceToIpMap[ifaceName]; ok {
+		return ip, nil
+	}
+	intf, err := net.InterfaceByName(ifaceName)
+	if err != nil {
+		return net.IPv4zero, err
+	}
+	addrs, err := intf.Addrs()
+	if err != nil {
+		return net.IPv4zero, err
+	}
+	for _, addr := range addrs {
+		if ipv4Addr := addr.(*net.IPNet).IP.To4(); ipv4Addr != nil {
+			return ipv4Addr, nil
+		}
+	}
```

Comment:
I wonder if we want this part. If we can safely fall back to this, why not to read this address at program load time :thinking: I am not sure. I guess it does not hurt.

Reply:
ya I agree. We can just mark the interface dirty and return.

Reply:
imo no need to make it dirty, program would not apply until there is an IP. Perhaps would not even get here if no IP was considered as device removed. Will not having an IP trigger removal?